samples/tutorial/Tutorial2.md

```diff
@@ -522,102 +540,62 @@ object RootWorkflow : StatefulWorkflow<Unit, State, Nothing, Any>() {
-
-    // Our list of back stack items. Will always include the "WelcomeScreen".
-    val backstackScreens = mutableListOf<Any>()
-
-    // Render a child workflow of type WelcomeWorkflow. When renderChild is called, the
-    // infrastructure will create a child workflow with state if one is not already running.
-    val welcomeScreen = context.renderChild(WelcomeWorkflow) { output ->
-      // When WelcomeWorkflow emits LoggedIn, turn it into our login action.
-      login(output.username)
+  ): BackStackScreen<*> {
+    // We always render the welcomeScreen regardless of the current state.
+    // It's either showing or else we may want to pop back to it.
+    val welcomeScreen = context.renderChild(WelcomeWorkflow) { loggedIn ->
+      // When WelcomeWorkflow emits LoggedIn, enqueue our log in action.
+      logIn(loggedIn.username)
     }
-    backstackScreens += welcomeScreen
 
-    when (renderState) {
-      // When the state is Welcome, defer to the WelcomeWorkflow.
-      is Welcome -> {
-        // We always add the welcome screen to the backstack, so this is a no op.
+    return when (renderState) {
+      is ShowingWelcome -> {
+        BackStackScreen(welcomeScreen)
       }
 
-      // When the state is Todo, defer to the TodoListWorkflow.
-      is Todo -> {
-        val todoListScreen = context.renderChild(TodoListWorkflow, props = ListProps(renderState.username)) {
-          // When receiving a Back output, treat it as a logout action.
-          logout()
+      is ShowingTodo -> {
+        val todoBackStack = context.renderChild(
+          child = TodoNavigationWorkflow,
+          props = TodoProps(renderState.username)
+        ) {
+          // When TodoNavigationWorkflow emits Back, enqueue our log out action.
+          logOut
         }
-        backstackScreens.add(todoListScreen)
+        listOf(welcomeScreen, todoBackStack).toBackStackScreen()
       }
     }
-
-    // Finally, return the BackStackScreen with a list of BackStackScreen.Items
-    return backstackScreens.toBackStackScreen()
   }
-
-  // …
-}
 ```
 
-We also need to add `BackStackContainer` to the view registry to actually wire up the transition animations:
+We also need to update the output type of `TodoListWorkflow`.
+So far it has output `Nothing`.
+Define a `BackPressed` object and use it as `TodoListWorkflow`'s output type.
+Update `render()` to create an `eventHandler` function to post the new output event.
 
-```kotlin
-// TutorialActivity.kt
-
-private val viewRegistry = ViewRegistry(
-    BackStackContainer,
-    WelcomeLayoutRunner,
-    TodoListLayoutRunner
-)
-```
-
-We also need to update the output type of `TodoListWorkflow`. So far it has output `Nothing`. Define a `Back` class, update `TodoListWorkflow`'s output type, then return this output inside `onBack`:
+At the same time, update TodoListScreen with an `onBack` event handler,
+and use workflow's handy `View.setBackHandler` function to respond to Android back press events.
```

Comment:
Add a note about `setBackHandler`, that it's compatible with androidX and Compose conventions.